sgt-back-end/index.js

```diff
@@ -0,0 +1,116 @@
+const express = require('express');
+const pg = require('pg');
+const app = express();
+const jsonParse = express.json();
+app.use('/api/grades', jsonParse);
+
+const db = new pg.Pool({
+  connectionString: 'postgres://dev:dev@localhost/studentGradeTable',
+  ssl: {
+    rejectUnauthorized: false
+  }
+});
+
+app.get('/api/grades', (req, res, next) => {
+  const sql = `
+    select *
+      from "grades"
+  `;
+  db.query(sql).then(result => {
+    const grades = result.rows;
+    res.json(grades);
+  }).catch(err => {
+    console.error(err);
+    res.status(500).json({ error: 'An unexpected error occured' });
+  });
+});
+
+app.post('/api/grades/', (req, res, next) => {
+  const name = req.body;
+  const course = req.body;
+  const score = req.body;
+  const text = 'INSERT INTO "grades"("name", "course", "score") VALUES($1, $2, $3) RETURNING *';
+  const values = [name, course, score];
+  if (name === undefined || course === undefined || parseInt(score) < 0 || parseInt(score) > 100) {
+    res.status(400).json({
+      error: 'Invalid field submitted'
```

Comment:
When the client sends the wrong data we want to give them information on how they can fix the request, so being a bit more detailed with what was wrong is helpful.